docs/src/format.md

```diff
@@ -0,0 +1,219 @@
+
+```hdf5
+parameters.hdf5
+└───layerId{1} (group)
+│   ├── arrayId{1}
+│   └── arrayId{2}
+└───layerId{2} (group)
+    ├── arrayId{1}
+    └── arrayId{2}
+```
+
+The indexing convention and naming for `arrayId` depend on the neural network model library:
+
+- Neural network models in the PEtab SciML [YAML format](@ref YAML_net_format) follow PyTorch indexing and naming conventions. For example, in a PyTorch `linear` layer, the arrays ids are `weight` and (optionally) `bias`
+- Neural network models in other formats follow the indexing and naming conventions of the respective package and programming language.
+
+### [Neural Network Input](@id hdf5_input_structure)
+
+Potential neural network input files are expected to have the following format:
+
+```hdf5
+input.hdf5
+└───input (group)
+    └─── input_array
+```
+
+As with [parameters](@ref hdf5_ps_structure), the indexing depends on the neural network library:
+
+- Neural network models in the PEtab SciML [YAML format](@ref YAML_net_format) follow PyTorch indexing. For example, if the first layer is `Conv2d`, the input should be in `(C, W, H)` format.
+- Neural network models in other formats follow the indexing and naming conventions of the respective package and programming language.
+
+!!! tip "For developers: Respect memory order"
+    Tools supporting the SciML extension should, for computational efficiency, reorder input data and potential layer parameter arrays to match the memory ordering of the target language. For example, PEtab.jl converts input data to follow Julia based indexing.
+
+TODO: We will fix condition specific input in the YAML file later.
+
+### [YAML Network file format](@id YAML_net_format)
+
+The `petab_sciml` library provides a YAML neural network file format for model exchange. The YAML format follows PyTorch conventions for layer names and arguments. While YAML network files can be written manually, it is recommended approach to define a PyTorch `nn.Module` and use the `petab_sciml` library to automatically generate the YAML representation (see tutorials).
+
+TODO: Should we have some description of the YAML format here? (can and should probably be added later)
+
+## [Mapping Table](@id mapping_table)
+
+All neural networks are assigned an Id in the PEtab problem [YAML](@ref YAML_file) file. A neural network Id is not considered a valid PEtab identifier, to avoid confusion about what it refers to (e.g., parameters, inputs, outputs). Consequently, every neural network input, parameter, and output referenced in the PEtab problem must be defined under `modelEntityId` and mapped to a PEtab identifier. For the `PEtabEntityId` column the same rules as in PEtab v2 apply, and additionally array file Ids defined in the [YAML](@ref YAML_file) file are considered valid PEtab entities.

```

Comment:
```suggestion
All neural networks are assigned an identifier in the PEtab problem [YAML](@ref YAML_file) file. A neural network identifier is not considered a valid PEtab identifier, to avoid confusion about what it refers to (e.g., parameters, inputs, outputs). Consequently, every neural network input, parameter, and output referenced in the PEtab problem must be defined under `modelEntityId` and mapped to a PEtab identifier. For the `PEtabEntityId` column the same rules as in PEtab v2 apply, and additionally array file Ids defined in the [YAML](@ref YAML_file) file are considered valid PEtab entities.

```